lib/qa_topic_edits.rb

```diff
@@ -61,10 +61,32 @@ class ::Topic
 
   def self.voted(topic, user)
     return nil if !user || !SiteSetting.qa_enabled
+    return nil if !self.can_vote(topic, user)
+  end
+
+  def self.can_vote(topic, user)
+    return nil if !user || !SiteSetting.qa_enabled
+    vote_count = self.vote_count(topic, user)
+
+    case user.trust_level
+    when 2
+        vote_limit = SiteSetting.qa_tl2_vote_limit
+    when 3
+        vote_limit = SiteSetting.qa_tl3_vote_limit
+    when 4
+        vote_limit = SiteSetting.qa_tl4_vote_limit
+    else
+        vote_limit = SiteSetting.qa_tl1_vote_limit
+    end
+    return vote_limit < vote_count
```

Comment:
Ruby will return the output of the final line of a method without a return statement, and the convention is to not use a return, e.g.
```
vote_limit < vote_count
```

Reply:
Neat.